src/performance/metrics/PerformanceUtil.cpp

```diff
@@ -1115,92 +661,16 @@ void PerformanceUtil::initiateCollectingRemoteSLAResourceUtilization(std::string
-
-    if (sockfd < 0) {
-        std::cerr << "Cannot create socket" << std::endl;
-        return 0;
-    }
-
-    if (host.find('@') != std::string::npos) {
-        host = Utils::split(host, '@')[1];
-    }
-
-    server = gethostbyname(host.c_str());
-    if (server == NULL) {
-        scheduler_logger.error("ERROR, no host named " + host);
-        return 0;
-    }
-
-    bzero((char *)&serv_addr, sizeof(serv_addr));
-    serv_addr.sin_family = AF_INET;
-    bcopy((char *)server->h_addr, (char *)&serv_addr.sin_addr.s_addr, server->h_length);
-    serv_addr.sin_port = htons(port);
-    if (Utils::connect_wrapper(sockfd, (struct sockaddr *)&serv_addr, sizeof(serv_addr)) < 0) {
-        std::cerr << "ERROR connecting" << std::endl;
-        return 0;
-    }
-
-    bzero(data, 301);
-    write(sockfd, JasmineGraphInstanceProtocol::HANDSHAKE.c_str(), JasmineGraphInstanceProtocol::HANDSHAKE.size());
-    scheduler_logger.log("Sent : " + JasmineGraphInstanceProtocol::HANDSHAKE, "info");
-    bzero(data, 301);
-    read(sockfd, data, 300);
-    string response = (data);
-
-    response = Utils::trim_copy(response);
-
-    if (response.compare(JasmineGraphInstanceProtocol::HANDSHAKE_OK) == 0) {
-        scheduler_logger.log("Received : " + JasmineGraphInstanceProtocol::HANDSHAKE_OK, "info");
-        write(sockfd, masterIP.c_str(), masterIP.size());
-        scheduler_logger.log("Sent : " + masterIP, "info");
-
-        bzero(data, 301);
-        read(sockfd, data, 300);
-        response = (data);
-        response = Utils::trim_copy(response);
-        scheduler_logger.log("Received : " + JasmineGraphInstanceProtocol::OK, "info");
-
-        if (response.compare(JasmineGraphInstanceProtocol::HOST_OK) == 0) {
-            write(sockfd, JasmineGraphInstanceProtocol::REQUEST_COLLECTED_STATS.c_str(),
-                  JasmineGraphInstanceProtocol::REQUEST_COLLECTED_STATS.size());
-            scheduler_logger.log("Sent : " + JasmineGraphInstanceProtocol::REQUEST_COLLECTED_STATS, "info");
-            bzero(data, 301);
-            read(sockfd, data, 300);
-            response = (data);
-            response = Utils::trim_copy(response);
-            string status = response.substr(response.size() - 5);
-            std::string result = response.substr(0, response.size() - 5);
-
-            while (status == "/SEND") {
-                write(sockfd, status.c_str(), status.size());
-
-                bzero(data, 301);
-                read(sockfd, data, 300);
-                response = (data);
-                response = Utils::trim_copy(response);
-                status = response.substr(response.size() - 5);
-                std::string loadAverageString = response.substr(0, response.size() - 5);
-                result = result + loadAverageString;
-            }
-            response = result;
-        }
-    }
-
-    return response;
+    // TODO(thevindu-w): master should not request load averages directly from workers.
```

Comment:
Can we please implement this missing feature?

Reply:
It's not actually a TODO. It says the master should not directly request performance metrics from workers.

Reply:
Then shall we remove labeling TODO. It will be misleading to have it marked as a TODO.

Reply:
Removed TODO. But the comment description is kept for future reference.

Reply:
Ok